File: csharp/SecureApp/Api/Models/Task.cs
// File: Api/Models/Task.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Api.Models
{
   [Table("Tasks")]
   public class Task
   {
      [Key]
      [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
      public long Id { get; set; }

      [Required]
      [MaxLength(200)]
      public string Name { get; set; } = string.Empty;

      [MaxLength(2000)]
      public string? Description { get; set; }

      [Required]
      [MaxLength(100)]
      public string CreatedBy { get; set; } = string.Empty;

      [MaxLength(100)]
      public string? AssignTo { get; set; }

      [Required]
      public TaskStatus Status { get; set; } = TaskStatus.Todo;

      [Required]
      public DateTime CreateDate { get; set; }

      public void OnCreate()
      {
         if (CreateDate == default)
         {
            CreateDate = DateTime.UtcNow;
         }

         if (Status == default)
         {
            Status = TaskStatus.Todo;
         }
      }
   }
}

File: java/spring_with_security/src/main/java/com/example/demo/model/Task.java
// src/main/java/com/example/demo/model/Task.java
package com.example.demo.model;

import jakarta.persistence.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "tasks")
public class Task {

   @Id
   @GeneratedValue(strategy = GenerationType.IDENTITY)
   @Column(name = "id")
   private Long id;

   @Column(name = "name")
   private String name;

   @Column(name = "description", length = 2000)
   private String description;

   @Column(name = "created_by", updatable=false)
   private String createdBy;

   @Column(name = "assign_to")
   private String assignTo;

   @Enumerated(EnumType.STRING)
   @Column(name = "status")
   private TaskStatus status;

   
   @Column(name = "create_date", updatable=false)
   private LocalDateTime createDate;

   @PrePersist
   public void onCreate() {
      if (createDate == null) {
         createDate = LocalDateTime.now();
      }
      if (status == null) {
         status = TaskStatus.TODO;
      }
   }

   // getters and setters...
   public void setId(Long id) {
      this.id = id;
   }

   // Add this getter method
   public Long getId() {
      return id;
   }

   public String getName() {
      return name;
   }

   public String getDescription() {
      return description;
   }

   public String getCreatedBy() {
      return createdBy;
   }

   public String getAssignTo() {
      return assignTo;
   }

   public TaskStatus getStatus() {
      return status;
   }

   public LocalDateTime getCreateDate() {
      return createDate;
   }

   // Add these to your Task.java
   public void setName(String name) {
      this.name = name;
   }

   public void setDescription(String description) {
      this.description = description;
   }

   public void setCreatedBy(String createdBy) {
      this.createdBy = createdBy;
   }

   public void setAssignTo(String assignTo) {
      this.assignTo = assignTo;
   }

   public void setStatus(TaskStatus status) {
      this.status = status;
   }

   public void setCreateDate(LocalDateTime createDate) {
      this.createDate = createDate;
   }
}